jenkins/databricks/build.sh

```diff
@@ -15,437 +15,292 @@
-   -DartifactId=parquet-hadoop \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$PARQUETCOMMONJAR \
-   -DgroupId=org.apache.parquet \
-   -DartifactId=parquet-common \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$PARQUETCOLUMNJAR \
-   -DgroupId=org.apache.parquet \
-   -DartifactId=parquet-column \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$PARQUETFORMATJAR \
-   -DgroupId=org.apache.parquet \
-   -DartifactId=parquet-format \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ARROWFORMATJAR \
-   -DgroupId=org.apache.arrow \
-   -DartifactId=arrow-format \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ARROWMEMORYJAR \
-   -DgroupId=org.apache.arrow \
-   -DartifactId=arrow-memory \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ARROWVECTORJAR \
-   -DgroupId=org.apache.arrow \
-   -DartifactId=arrow-vector \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ORC_CORE_JAR \
-   -DgroupId=org.apache.orc \
-   -DartifactId=orc-core \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ORC_SHIM_JAR \
-   -DgroupId=org.apache.orc \
-   -DartifactId=orc-shims \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ORC_MAPREDUCE_JAR \
-   -DgroupId=org.apache.orc \
-   -DartifactId=orc-mapreduce \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$PROTOBUF_JAR \
-   -DgroupId=com.google.protobuf \
-   -DartifactId=protobuf-java \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B -Ddatabricks -Dbuildver=$BUILDVER clean package -DskipTests $MVN_OPT
-
+## can be overwritten by shell variables, e.g. "BASE_SPARK_VERSION=3.1.2 MVN_OPT=bash build.sh"
+
+# default SPARK Version is extracted from DB Spark environment
+DEFAULT_SPARK_BASE_VERSION=$(< /databricks/spark/VERSION)
+
+# list of supported sparks
+SUPPORTED_SPARKS=("3.1.2" "3.2.1" "3.3.0")
```

Comment:
Why do we need this check? The build /test will fail if it's not supported anyways. If at all we should be checking runtime versions like 11.3

Reply:
This was the behavior in the original code. It fails if the base_spark is not "3.2.1/3.1.2".
I can get rid of it anyway.